taskcluster/app_services_taskgraph/transforms/__init__.py

```diff
@@ -24,14 +24,19 @@ def publications_to_artifact_paths(name, version, publications, secondary_extens
     return paths
 
 
-def publications_to_artifact_map_paths(name, version, publications, secondary_extensions):
+def publications_to_artifact_map_paths(name, version, publications, nightly_build, secondary_extensions):
     build_map_paths = {}
     for publication in publications:
         for extension in _extensions(publication["type"], secondary_extensions):
-            artifact_filename = _artifact_filename(publication['name'], version, extension)
+            publication_name = publication['name']
+            artifact_filename = _artifact_filename(publication_name, version, extension)
+            if nightly_build:
+                destination = "maven2/org/mozilla/appservices/nightlies/{}/{}/{}".format(publication_name, version, artifact_filename)
+            else:
+                destination = "maven2/org/mozilla/appservices/{}/{}/{}".format(publication_name, version, artifact_filename)
```

Comment:
Is this the way we should handle nightlies?  I used `geckoview` as the guide here.  It has a [different directory for each channel in maven](https://maven.mozilla.org/?prefix=maven2/org/mozilla/geckoview/).

I think other components use a different maven repo and/or package name.  Should we be doing that?

Should normal releases go into a separate directory as well?

Reply:
Good question! 

I know Android-Components are shipped on [nightly.maven.mozilla.org](https://nightly.maven.mozilla.org/?prefix=maven2/org/mozilla/components/). We've never moved geckoview there likely because nightlies.m.o arrived after we first shipped geckoview nightly.